integration-cli/docker_cli_pull_test.go

```diff
@@ -369,3 +369,40 @@ func (s *DockerTrustSuite) TestTrustedPullWithExpiredSnapshot(c *check.C) {
 		}
 	})
 }
+
+// Test that pull continues after client has disconnected. #15589
+func (s *DockerTrustSuite) TestPullClientDisconnect(c *check.C) {
```

Comment:
Actually, DockerSuite, since it doesn't involve a private registry either. Opened a PR here: https://github.com/docker/docker/pull/15894